enhancements/network/configurable-network-diagnostics-pod-placement.md

```diff
@@ -0,0 +1,303 @@
+allows the administrator to disable the network diagnostics. With this enhancement CNO would go 
+degraded if `networkDiagnostics` is configured when `disableNetworkDiagnostics` is set. 
+Once the new API lands in the default featureset, `disableNetworkDiagnostics` is going 
+to be deprecated in favor of `networkDiagnostics`.
+
+This enhancement adds the following fields to the spec of the `network.config.openshift.io` CRD:
+```
+type NetworkSpec struct {
+...
+    // networkDiagnostics defines network diagnostics configuration
+    // +optional
+    NetworkDiagnostics NetworkDiagnostics `json:"networkDiagnostics"`
+}
+
+// NetworkDiagnosticsMode is an enumeration of the available network diagnostics modes
+// Valid values are "All", "Disabled".
+// +kubebuilder:validation:Enum:=All;Disabled
+type NetworkDiagnosticsMode string
+
+const (
+	// NetworkDiagnosticsAll meants that all network diagnostics checks are enabled
+	NetworkDiagnosticsAll  NetworkDiagnosticsMode = "All"
+	// NetworkDiagnosticsDisabled means that network diagnostics checks are disabled
+	NetworkDiagnosticsDisabled NetworkDiagnosticsMode = "Disabled"
+)
+
+// NetworkDiagnostics defines network diagnostics configuration
+type NetworkDiagnostics struct {
+	// mode controls the network diagnostics mode
+	//
+	// By default the value is set to All.
+	//
+	// +optional
+	// +kubebuilder:default=All
+	Mode NetworkDiagnosticsMode `json:"mode"`
+
+	// sourcePlacement controls the scheduling of network diagnostics source deployment
+	//
+	// See NetworkDiagnosticsNodePlacement for more details about default values.
+	//
+	// +optional
+	SourcePlacement NetworkDiagnosticsNodePlacement `json:"sourcePlacement"`
+
+	// targetPlacement controls the scheduling of network diagnostics target daemonset
+	//
+	// See NetworkDiagnosticsNodePlacement for more details about default values.
+	//
+	// +optional
+	TargetPlacement NetworkDiagnosticsNodePlacement `json:"targetPlacement"`
+}
+
+// NetworkDiagnosticsNodePlacement defines node scheduling configuration network diagnostics components
+type NetworkDiagnosticsNodePlacement struct {
+	// nodeSelector is the node selector applied to a network diagnostics component
+	//
+	// By default this is set to `kubernetes.io/os: linux`
+	//
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector"`
+
+	// tolerations is a list of tolerations applied to a network diagnostics component
+	//
+	// For sourcePlacement, this is set to an empty list by default.
+	//
+	// For targetPlacement, this is set to `- operator: "Exists"` by default.
+	//
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations"`
+}
+```
+The fields will initially be added as tech preview and will move to default once they meet the 
+graduation criteria.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+In Hypershift, the network diagnostics components do not run in the controlplane so there are no 
```

Comment:
For hosted cluster,  configuring the network diagnostics pod placement also work, right?

Reply:
Yes, this is not different for HyperShift.
One caveat is whether the HyperShift operator allows modifying the network.config CR from the hosted cluster.